save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt

```diff
@@ -408,17 +403,24 @@ internal class ContestController(
         }
         .map { (organization, contests) ->
             contestsRequest.map { contestRequest ->
-                contestService.updateContest(
-                    contestRequest.toContest(organization).apply { id = contests.single { name == it.get().name }.get().id }
-                )
+                contests.single {
+                    contestRequest.name == it.get().name && contestRequest.organizationName == it.get().organization.name
+                }
+                    .get()
+                    .let { contest ->
+                        contestService.updateContest(
+                            contestRequest.toContest(organization, contest.testSuiteLinks).apply {
+                                id = contest.id
+                            }
+                        )
+                    }
             }
             ResponseEntity.ok("Contest successfully updated")
         }
 
-    private fun getContestOrNotFound(contestName: String): Mono<Contest> = Mono.just(contestName)
-        .flatMap {
-            Mono.justOrEmpty(contestService.findByName(it))
-        }
+    private fun getContestOrNotFound(contestName: String): Mono<Contest> = Mono.justOrEmpty(
+        contestService.findByName(contestName)
```

Comment:
now it's not a lazy calculation. can we change it to `Mono.fromCallable` or `blockingToMono`

Reply:
Changed

Reply:
Fixed